json/src/test/java/org/apache/JsonGenericSerializerTest.java

```diff
@@ -64,6 +64,14 @@ void shouldThrowWhenDeserializeEventWithMissingType() {
             .isInstanceOf(JsonGenericSerializer.InvalidTypeException.class);
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
it looks like there are more than one path in the method under test. You probably lack some tests, don't you?